src/api/mod.rs

```diff
@@ -253,66 +252,59 @@ pub enum WriteType {
     WithoutResponse,
 }
 
+#[async_trait]
```

Comment:
nit: The attribute would it just above the pub trait declaration, instead of above the doc comments. to make it easier to identify in the event the documentation grows to be quite large

Reply:
Done.